src/login/logind.c

```diff
@@ -410,74 +411,86 @@ static int parse_fdname(const char *fdname, char **session_id, dev_t *dev) {
         return 0;
 }
 
-static int manager_attach_fds(Manager *m) {
-        _cleanup_strv_free_ char **fdnames = NULL;
-        int n, i, fd;
+/* Release stored fd from PID 1.  Note DRM fds can hold a lot of memory.
+ * Let's at least implement this without requiring a heap allocation :). */
 
-        /* Upon restart, PID1 will send us back all fds of session devices
-         * that we previously opened. Each file descriptor is associated
-         * with a given session. The session ids are passed through FDNAMES. */
+static void fdstore_remove(const char *fdname) {
+        char message[sizeof("FDSTOREREMOVE=1\n" "FDNAME=\n") + 255];
 
-        n = sd_listen_fds_with_names(true, &fdnames);
-        if (n <= 0)
-                return n;
+        xsprintf(message, "FDSTOREREMOVE=1\n" "FDNAME=%s\n", fdname);
```

Comment:
i think it would be easier to read if we'd format this so that a line break is between the two fields, mirroring the \n in the string...